wts/api.py

```diff
@@ -74,23 +75,40 @@ def load_settings(app):
         redirect_uri = get_var("REDIRECT_URI", default="", secret_config=conf)
         state_prefix = wts_hostname or ""
         if not redirect_uri:
-            redirect_uri = wts_base_url + "oauth2/authorize"
+            redirect_uri = urljoin(wts_base_url, "oauth2/authorize")
             state_prefix = ""
 
         for idp, idp_conf in conf.get("login_options", {}).items():
-            authorization_url = fence_base_url + "oauth2/authorize"
-            authorization_url = add_params_to_uri(
-                authorization_url, idp_conf.get("params", {})
-            )
+            scope = "openid data user"
+            idp_params = idp_conf.get("params", {})
+            idp_client = idp_params.get("idp", "")
+
+            if "auth_url" in idp_params and "token_url" in idp_params:  # Generic OIDC
+                auth_endpoint = idp_params.get("auth_url")
+                token_endpoint = idp_params.get("token_url")
+                username_field = idp_params.get("id_token_username_field", "")
+                scope = idp_params.get("scope", scope)
+                authorization_url = urljoin(url, auth_endpoint)
+                access_token_url = urljoin(url, token_endpoint)
```

Comment:
the changes lgtm. but i think we could use unit tests to cover the different client config and the id_token_username_field setting

Reply:
Ok, I'll make a new unit test for the different config